[32mINFO    [0m pytest_operator.plugin:plugin.py:834 Model status:

Model  Controller           Cloud/Region         Version  SLA          Timestamp
test   localhost-localhost  localhost/localhost  3.1.8    unsupported  01:54:04Z

App                  Version  Status  Scale  Charm                Channel  Rev  Exposed  Message
pgbouncer            1.21.0   error       3  pgbouncer                       0  no       hook failed: "backend-database-relation-changed"
postgresql           14.11    active      3  postgresql           14/edge  401  no       
postgresql-test-app           active      3  postgresql-test-app  edge     115  no       

Unit                    Workload     Agent      Machine  Public address  Ports     Message
postgresql-test-app/0   active       idle       0        10.39.172.111             
  pgbouncer/1           maintenance  executing           10.39.172.111             updating PgBouncer config
postgresql-test-app/1   active       executing  1        10.39.172.250             
  pgbouncer/2           error        idle                10.39.172.250             hook failed: "backend-database-relation-changed"
postgresql-test-app/2*  active       idle       2        10.39.172.165             
  pgbouncer/0*          active       executing           10.39.172.165             
postgresql/0            active       executing  3        10.39.172.157   5432/tcp  
postgresql/1            active       executing  4        10.39.172.28    5432/tcp  
postgresql/2*           active       executing  5        10.39.172.208   5432/tcp  Primary

Machine  State    Address        Inst id        Base          AZ  Message
0        started  10.39.172.111  juju-1c4e6b-0  ubuntu@22.04      Running
1        started  10.39.172.250  juju-1c4e6b-1  ubuntu@22.04      Running
2        started  10.39.172.165  juju-1c4e6b-2  ubuntu@22.04      Running
3        started  10.39.172.157  juju-1c4e6b-3  ubuntu@22.04      Running
4        started  10.39.172.28   juju-1c4e6b-4  ubuntu@22.04      Running
5        started  10.39.172.208  juju-1c4e6b-5  ubuntu@22.04      Running

[32mINFO    [0m pytest_operator.plugin:plugin.py:840 Juju error logs:

machine-2: 01:38:24 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-2: 01:38:24 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-2: 01:38:24 ERROR juju.worker.dependency "broker-tracker" manifold worker returned unexpected error: no container types determined
unit-postgresql-test-app-2: 01:38:24 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-0: 01:38:24 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-0: 01:38:24 ERROR juju.worker.dependency "broker-tracker" manifold worker returned unexpected error: no container types determined
machine-0: 01:38:24 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-test-app-0: 01:38:24 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
unit-pgbouncer-0: 01:38:28 ERROR juju.worker.meterstatus error running "meter-status-changed": exit status 1
unit-pgbouncer-1: 01:38:29 ERROR juju.worker.meterstatus error running "meter-status-changed": exit status 1
machine-1: 01:39:08 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-1: 01:39:08 ERROR juju.worker.dependency "broker-tracker" manifold worker returned unexpected error: no container types determined
machine-1: 01:39:08 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-test-app-1: 01:39:08 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
unit-pgbouncer-2: 01:39:13 ERROR juju.worker.meterstatus error running "meter-status-changed": exit status 1
machine-5: 01:41:50 ERROR juju.worker.dependency "broker-tracker" manifold worker returned unexpected error: no container types determined
machine-5: 01:41:50 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-5: 01:41:50 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-2: 01:41:51 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-3: 01:41:56 ERROR juju.worker.dependency "broker-tracker" manifold worker returned unexpected error: no container types determined
machine-3: 01:41:56 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-3: 01:41:56 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-0: 01:41:57 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-4: 01:41:57 ERROR juju.worker.dependency "broker-tracker" manifold worker returned unexpected error: no container types determined
machine-4: 01:41:57 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-4: 01:41:57 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-1: 01:41:58 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
unit-pgbouncer-2: 01:54:03 ERROR unit.pgbouncer/2.juju-log backend-database:7: Uncaught exception while in charm code:
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-pgbouncer-2/charm/venv/ops/model.py", line 3022, in _run
    result = subprocess.run(args, **kwargs)  # type: ignore
  File "/usr/lib/python3.10/subprocess.py", line 526, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '('/var/lib/juju/tools/unit-pgbouncer-2/secret-get', '--label', 'pgb-peers.pgbouncer.app', '--format=json')' returned non-zero exit status 1.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-pgbouncer-2/charm/./src/charm.py", line 808, in <module>
    main(PgBouncerCharm)
  File "/var/lib/juju/agents/unit-pgbouncer-2/charm/venv/ops/main.py", line 544, in main
    manager.run()
  File "/var/lib/juju/agents/unit-pgbouncer-2/charm/venv/ops/main.py", line 520, in run
    self._emit()
  File "/var/lib/juju/agents/unit-pgbouncer-2/charm/venv/ops/main.py", line 509, in _emit
    _emit_charm_event(self.charm, self.dispatcher.event_name)
  File "/var/lib/juju/agents/unit-pgbouncer-2/charm/venv/ops/main.py", line 143, in _emit_charm_event
    event_to_emit.emit(*args, **kwargs)
  File "/var/lib/juju/agents/unit-pgbouncer-2/charm/venv/ops/framework.py", line 352, in emit
    framework._emit(event)
  File "/var/lib/juju/agents/unit-pgbouncer-2/charm/venv/ops/framework.py", line 851, in _emit
    self._reemit(event_path)
  File "/var/lib/juju/agents/unit-pgbouncer-2/charm/venv/ops/framework.py", line 941, in _reemit
    custom_handler(event)
  File "/var/lib/juju/agents/unit-pgbouncer-2/charm/src/relations/backend_database.py", line 226, in _on_relation_changed
    if not self.charm.check_pgb_running():
  File "/var/lib/juju/agents/unit-pgbouncer-2/charm/./src/charm.py", line 464, in check_pgb_running
    if self.backend.ready:
  File "/var/lib/juju/agents/unit-pgbouncer-2/charm/src/relations/backend_database.py", line 431, in ready
    not (auth_file := self.charm.get_secret(APP_SCOPE, AUTH_FILE_DATABAG_KEY))
  File "/var/lib/juju/agents/unit-pgbouncer-2/charm/./src/charm.py", line 286, in get_secret
    return self.peer_relation_data(scope).get_secret(peers.id, secret_key)
  File "/var/lib/juju/agents/unit-pgbouncer-2/charm/lib/charms/data_platform_libs/v0/data_interfaces.py", line 519, in wrapper
    return f(self, *args, **kwargs)
  File "/var/lib/juju/agents/unit-pgbouncer-2/charm/lib/charms/data_platform_libs/v0/data_interfaces.py", line 1871, in get_secret
    and full_field not in self.current_secret_fields
  File "/var/lib/juju/agents/unit-pgbouncer-2/charm/lib/charms/data_platform_libs/v0/data_interfaces.py", line 1832, in current_secret_fields
    if content := self._get_group_secret_contents(relation, group):
  File "/var/lib/juju/agents/unit-pgbouncer-2/charm/lib/charms/data_platform_libs/v0/data_interfaces.py", line 2066, in _get_group_secret_contents
    result = super()._get_group_secret_contents(relation, group, secret_fields)
  File "/var/lib/juju/agents/unit-pgbouncer-2/charm/lib/charms/data_platform_libs/v0/data_interfaces.py", line 996, in _get_group_secret_contents
    if (secret := self._get_relation_secret(relation.id, group)) and (
  File "/var/lib/juju/agents/unit-pgbouncer-2/charm/lib/charms/data_platform_libs/v0/data_interfaces.py", line 506, in wrapper
    return f(self, *args, **kwargs)
  File "/var/lib/juju/agents/unit-pgbouncer-2/charm/lib/charms/data_platform_libs/v0/data_interfaces.py", line 2055, in _get_relation_secret
    return self.secrets.get(label, secret_uri, legacy_labels=self._previous_labels())
  File "/var/lib/juju/agents/unit-pgbouncer-2/charm/lib/charms/data_platform_libs/v0/data_interfaces.py", line 702, in get
    if secret.meta:
  File "/var/lib/juju/agents/unit-pgbouncer-2/charm/lib/charms/data_platform_libs/v0/data_interfaces.py", line 605, in meta
    self._secret_meta = self._model.get_secret(label=label)
  File "/var/lib/juju/agents/unit-pgbouncer-2/charm/venv/ops/model.py", line 280, in get_secret
    content = self._backend.secret_get(id=id, label=label)
  File "/var/lib/juju/agents/unit-pgbouncer-2/charm/venv/ops/model.py", line 3370, in secret_get
    result = self._run('secret-get', *args, return_output=True, use_json=True)
  File "/var/lib/juju/agents/unit-pgbouncer-2/charm/venv/ops/model.py", line 3024, in _run
    raise ModelError(e.stderr) from e
ops.model.ModelError: ERROR permission denied

unit-pgbouncer-2: 01:54:03 ERROR juju.worker.uniter.operation hook "backend-database-relation-changed" (via hook dispatching script: dispatch) failed: exit status 1

[32mINFO    [0m pytest_operator.plugin:plugin.py:904 Forgetting main...